src/main/java/use_cases/arr/Add.java

```diff
@@ -17,7 +21,8 @@ public Add(ARROutputBoundary presenter){
      */
     public void changeItemQuantity(ARRInputData data){
 
-        if (data.getQuantity() < 0){
+        if (data.getQuantity() < 0 && !(this instanceof RemoveUseCase)) // To allow negative quantity values only when the item is being removed
+        {
```

Comment:
Initial if statement might be redundant and can be removed. Keep 2nd if statement.